### PR TITLE
Add architecture for DBT-2

### PR DESCRIPTION
**Pull request by mw2q**
Currently implemented only for AWS.

### CODE REVIEW

**Comment by jt-edb** [comment]
Could you please rebase your branch? Thank you

**Comment by mw2q** [comment]
> Could you please rebase your branch? Thank you

Rebased. :)

File: edbdeploy/spec/reference_architecture.py [hunk 1]
```diff
@@ -11,6 +11,11 @@
         'pooler_local': False,
         'efm': False,
         'replication_type': None,
+        'dbt2': False,
+        'dbt2_client': False,
+        'dbt2_client_count': 1,
```
Comment (jt-edb) [inline edbdeploy/spec/reference_architecture.py:16]: I'd suggest to set the default DBT2 client count to 0 here.

File: edbdeploy/spec/reference_architecture.py [hunk 2]
```diff
@@ -11,6 +11,11 @@
...
+        'dbt2_driver': False,
+        'dbt2_driver_count': 1,
```
Comment (jt-edb) [inline edbdeploy/spec/reference_architecture.py:18]: Same comment about the default count value.

File: edbdeploy/spec/reference_architecture.py [hunk 3]
```diff
@@ -26,6 +31,11 @@
         'pooler_local': False,
         'efm': True,
         'replication_type': "synchronous",
+        'dbt2': False,
+        'dbt2_client': False,
+        'dbt2_client_count': 1,
```
Comment (jt-edb) [inline edbdeploy/spec/reference_architecture.py:36]: Same comment about the default count value.

File: edbdeploy/spec/reference_architecture.py [hunk 4]
```diff
@@ -26,6 +31,11 @@
...
+        'dbt2_driver': False,
+        'dbt2_driver_count': 1,
```
Comment (jt-edb) [inline edbdeploy/spec/reference_architecture.py:38]: Same comment about the default count value.

File: edbdeploy/spec/reference_architecture.py [hunk 5]
```diff
@@ -41,6 +51,31 @@
         'pooler_local': False,
         'efm': True,
         'replication_type': "synchronous",
+        'dbt2': False,
+        'dbt2_client': False,
+        'dbt2_client_count': 1,
```
Comment (jt-edb) [inline edbdeploy/spec/reference_architecture.py:56]: Same comment about the default count value.

File: edbdeploy/spec/reference_architecture.py [hunk 6]
```diff
@@ -41,6 +51,31 @@
...
+        'dbt2_driver': False,
+        'dbt2_driver_count': 1,
```
Comment (jt-edb) [inline edbdeploy/spec/reference_architecture.py:58]: Same comment about the default count value.

**Review by jt-edb — COMMENTED** [review]
Few minor comments. Thank you.

**Comment by mw2q** [comment]
Oops, changed counts to 0 where appropriate.

**Review by jt-edb — APPROVED** [review]
LGTM

**Pull request merged**